### PR TITLE
feat(issue-details): Improve event selector UI

### LINKED ISSUE #45649: Issue details event selector updates
Background:

New Sentry users often don’t understand that the issue details page is a carousel, full of lots of events inside. Browsing events is a key part of the debugging experience, so we should do a better job of highlighting this feature.

Another issue we've been facing is the growing number of actions and link-outs that have been added to the event header. A new design should be able to accommodate these new features.

Solution:

- Add a new carousel component to combine the event ID with the previous and next buttons.
- To the right of the carousel, add an overflow menu for actions like downloading JSON and copying the event URL.
- Use a relative time for the event instead of a full absolute timestamp. The absolute time should still be viewable on hover. 

Image

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/sentry/issues/45649

Before:

CleanShot 2023-03-13 at 17 21 29

After:

CleanShot 2023-03-15 at 15 21 40

This also solves a couple existing issues with the current component:

- Performance issues were always displaying an incorrect date due to not having a `dateCreated` - this uses the `dateReceived` as a fallback
- The old component was not shown on small screens - we now show this on all screen sizes

### CODE REVIEW

**Comment by scefali** [comment]
@malwilley so we don't use the full ID in the view anymore?

File: static/app/views/issueDetails/groupEventCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+          icon: <IconEllipsis size={BUTTON_ICON_SIZE} />,
+          showChevron: false,
+          size: BUTTON_SIZE,
+        }}
+        items={[
+          {
+            key: 'copy-event-url',
+            label: t('Copy Event Link'),
+            onAction: () =>
+              copyToClipboard(
```
Comment (scefali) [inline static/app/views/issueDetails/groupEventCarousel.tsx:155]: @malwilley let's get some analytics on the actions like copying

File: static/app/views/issueDetails/groupEventCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  const groupId = group.id;
+  const hasReplay = Boolean(event?.tags?.find(({key}) => key === 'replayId')?.value);
+  const isReplayEnabled = organization.features.includes('session-replay');
+  const baseEventsPath = `/organizations/${organization.slug}/issues/${groupId}/events/`;
+  const jsonUrl = `/api/0/projects/${organization.slug}/${projectSlug}/events/${event.id}/json/`;
+  const latencyThreshold = 30 * 60 * 1000; // 30 minutes
+  const isOverLatencyThreshold =
+    event.dateReceived &&
+    event.dateCreated &&
+    Math.abs(+moment(event.dateReceived) - +moment(event.dateCreated)) > latencyThreshold;
```
Comment (scefali) [inline static/app/views/issueDetails/groupEventCarousel.tsx:70]: @malwilley what's up with the `+moment`?

File: static/app/views/issueDetails/groupEventCarousel.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/views/issueDetails/groupEventCarousel.tsx:194]: Added analytics to all the actions in the overflow menu

**Comment by malwilley** [comment]
@scefali Removed the "copy event ID" action from the menu and added this "click to copy" behavior for the short event ID:

CleanShot 2023-03-14 at 10 56 54

**Comment by scefali** [comment]
Screen Shot 2023-03-14 at 1 07 52 PM

@malwilley this doesn't look great, we should have it take up the full width

**Comment by scefali** [comment]
Screen Shot 2023-03-14 at 1 08 54 PM

Also this

**Comment by malwilley** [comment]
> @malwilley this doesn't look great, we should have it take up the full width

Yeah I think you're right, had added that after talking to @robinrendle about it looking awkward with wide screens but yeah taking up the whole width is better for now.

> Also this

Did that happen while running with `http`? The navigator.clipboard only works in secure contexts (https://github.com/getsentry/sentry/pull/45438). Although seeing this we should probably modify that component to handle that error a bit better.

**Comment by evanpurkhiser** [comment]
Can we not just put the whole event ID in there? Looks like there's enough space?

**Comment by malwilley** [comment]
There is enough space (at wider viewports), but we're going with the shortened event ID to keep the UI as visually simple as possible. This is also in line with what Discover shows for events.

cc @robinrendle

**Review by scefali — APPROVED** [review]

**Comment by dcramer** [comment]
Aside the reason we have the full datetime here is because we got feedback over the years that people wanted quick access to _exactly_ when something happened, as it lets them correlate incidents.

**Comment by evanpurkhiser** [comment]
Agree that it would be nice to show the full date time if it's wide enough?

**Comment by robinrendle** [comment]
@dcramer @evanpurkhiser I get that but see how much easier it is to see _when_ an event happened? Click through a few events and now your brain doesn’t have to parse a long-ass string like `Jan 4 2023 4:00:00PM UST`. Also, if ya wanna know when an incident happened then that time should apply to the state change, not the event — New, Regressed, Escalating, etc. 

Perhaps we can have a toggle somewhere to permanently see the full time if a bunch of customers scream at us but for now I’m gonna die on this hill.

**Comment by dcramer** [comment]
@robinrendle in the 15 years since I've built Sentry, I've never once needed to use that paginator. I have used the datetime, however.

**Comment by DAcodedBEAT** [comment]
> Did that happen while running with `http`? The navigator.clipboard only works in secure contexts (#45438). Although seeing this we should probably modify that component to handle that error a bit better.

If possible, can Sentry's web UI either disable the click to copy button in insecure contexts and/or express the error in the UI, explaining how click to copy is unsupported when running sentry insecurely?  Likewise, it might be helpful to have this in the selfhosted docs, in case folks get caught up on this in the future.

If you want me to create Github Issues / PRs for this, let me know and I'll try to do the needful.  (I created the original issue around this not working https://github.com/getsentry/sentry/issues/44025 so I can totally do this again. 😄)

**Pull request merged**